src/figuresClasses.ts

```diff
@@ -1,19 +1,85 @@
 export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle',
+  color: 'red' | 'green' | 'blue',
 
+  getArea(): number
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape: Figure.shape;
 
+  constructor(
+    public color: Figure.color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.shape = 'triangle';
+
+    if (a <= 0
+            || b <= 0
+            || c <= 0
+            || a + b <= c
+            || b + c <= a
+            || c + a <= b
+    ) {
```

Comment:
combine logic and move to variable. Ex. `const isValidValue =  a > 0 && b > 0 && c > 0`. All values must be greater than zero, not just some one, right?

Reply:
|| a + b <= c
|| b + c <= a
|| c + a <= b
this lasts cant make as an variable, test wont pass than, I tried it didnt work to me. First 3 alright